activestorage/lib/active_storage/attached/many.rb

```diff
@@ -55,7 +55,7 @@ def attach(*attachables)
           false
         end
       else
-        record.public_send("#{name}=", (change&.attachables || blobs) + attachables.flatten)
+        record.public_send("#{name}=", blobs + attachables.flatten)
```

Comment:
@santib Very minor, but now that we assign the same value I think we can remove the duplication by extracting a method:

```ruby
def attach(*attachables)
  if record.persisted? && !record.changed?
    self.attachables = attachables
    if record.save
      record.public_send("#{name}")
    else
      false
    end
  else
    self.attachables = attachables
  end
end

private
  def attachables=(attachables)
    record.public_send("#{name}=", blobs + attachables.flatten)
  end
```

Reply:
@jorgemanrubia what do you think about

```ruby
    def attach(*attachables)
      record.public_send("#{name}=", blobs + attachables.flatten)

      return unless record.persisted? && !record.changed?

      if record.save
        record.public_send("#{name}")
      else
        false
      end
    end
```

not sure if my version can be affecting the implicit return value in some case?

Reply:
@santib I thought of extracting the common line too, but my concern was changing the return value as you suggest. The method always returns the list of attachables unless it can't be saved, in which case it returns false. I would try to leave that unchanged

Reply:
I noticed that [`record.public_send("#{name}=", ...)`](https://github.com/rails/rails/blob/2077524d5efc270658b27be4d8b7ec63e394f3f7/activestorage/lib/active_storage/attached/model.rb#L143-L161) returns a different value than [`record.public_send("#{name}")`](https://github.com/rails/rails/blob/2077524d5efc270658b27be4d8b7ec63e394f3f7/activestorage/lib/active_storage/attached/model.rb#L135-L136).  If we do care about the *truthy* return value, is there a reason to not be consistent about it?  e.g.:

```ruby
def attach(*attachables)
  record.public_send("#{name}=", blobs + attachables.flatten)
  return false if record.persisted? && !record.changed? && !record.save
  record.public_send("#{name}")
end
```

Reply:
Thanks @santib.

@jonathanhefner That suggestion is more compact but I think it makes it less clear that the method can save the record, with the `save` is placed at the end of the inline conditions. It's a minor detail in any case. For me the patch is good to go, if you want to merge. I think we need to kick the test suite again though, lots of errors that seem unrelated?

Reply:
@jonathanhefner oh good point about the return type. I had assumed those were the same. There is also a test enforcing [that type](https://github.com/rails/rails/blob/b77be3f427c9d4be092b2a488e6286546c779990/activestorage/test/models/attached/many_test.rb#L852-L851). Also good point on the symmetry of the methods 👍. I like your implementation after having played a bit more with it:

```ruby
def attach(*attachables)
  record.public_send("#{name}=", blobs + attachables.flatten)
  return false if record.persisted? && !record.changed? && !record.save
  record.public_send("#{name}")
end
```

@santib could you please make that last change 🙏? Sorry about the back and forth.

Reply:
Sure thing, thanks Jonathan for the improvement. The `attach` method looks so much better now